auto_update_tests.py

```diff
@@ -2,6 +2,8 @@
 
 import os, sys, subprocess, difflib
 
+os.environ['LD_LIBRARY_PATH'] = 'lib' # find our dynamic libraries
```

Comment:
Hmm, seems more complicated? In particular we need that for the example tests, and if I understand RPATH, it wouldn't help there (since we don't build them in cmake, we intentionally build them in  check.py to simulate what a user would do).